internal/ledger/store/badger_test.go

```diff
@@ -97,3 +97,81 @@ func TestBadgerDB_Set(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, expectedValue, value)
 }
+
+func TestBadgerDB_NewIterator(t *testing.T) {
+	t.Parallel()
+
+	tmpdir, err := ioutil.TempDir("", "bdb-test-")
+	defer os.RemoveAll(tmpdir)
+	assert.NoError(t, err)
+
+	db, err := NewBadgerDB(tmpdir)
+	require.NoError(t, err)
+
+	var (
+		commonScope      Scope
+		commonPrefix     []byte
+		randKeys         []testBadgerKey
+		unexpectedValues [][]byte
+		keys             []testBadgerKey
+		expectedValues   [][]byte
+		expected         map[string][]byte
+	)
+
+	expected = make(map[string][]byte)
+
+	const (
+		ArrayLength = 3
+	)
+
+	fuzz.New().NilChance(0).Fuzz(&commonScope)
+	fuzz.New().NilChance(0).NumElements(ArrayLength, ArrayLength).Fuzz(&commonPrefix)
+
+	ff := fuzz.New().NilChance(0).NumElements(ArrayLength, ArrayLength).Funcs(
+		func(key *testBadgerKey, c fuzz.Continue) {
+			var id []byte
+			c.Fuzz(&id)
+			*key = testBadgerKey{id: id, scope: commonScope}
```

Comment:
It's better to assign field here, not construct another value.
```
key.id = id
key.scope = commonScope
```

Reply:
done